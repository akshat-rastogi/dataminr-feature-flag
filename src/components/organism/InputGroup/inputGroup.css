.individual div .input-group,
.group, 
.toggle-group, .toggle-group div .input-group {
    border-radius: 0.4rem;
    overflow: hidden;
}
.toggle-group{
    background: rgb(71, 71, 71);
    padding-top: 3px;
    padding-bottom: 3px;
}
.toggle-group div .input-group{
    margin-right: 3px;
    margin-left: 3px;
}
.toggle-group > div:not(:last-child){
    margin-bottom: 3px;
}
.individual div .input-group{
    margin-bottom: 15px;
}
.input-group{
    background: #1a1a1a;
    padding: 10px;
}
.label{
    width:100%;
    text-transform: uppercase;
    font-size: 1rem;
}
.bullet{
    min-height: 20px;
    min-width: 20px;
    background: #686868;
    border-radius: 0.2rem;
    margin-right: 5px;
}
.display-flex{
    display: flex;
    align-items: center;
}
.input-addon{
    display: flex;
}
.childToggle{
    width: 50px;
    cursor: pointer;
    text-align: right;
}
.childToggle:after{
    content: '∨';
}
.childToggle.open:after{
    content: '∧';
}