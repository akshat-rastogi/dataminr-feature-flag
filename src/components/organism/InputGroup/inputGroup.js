import React from 'react';
import { ToggleSwitch } from '../../molecule/ToggleSwitch';
import { useState } from 'react';
import './inputGroup.css';
import { NumericDropdown } from '../../atom/NumericDropdown';

const InputGroup = (props) => {
    const [value, setValue] = useState(props.defaultValue || false);
    const [isOpen, setIsOpen] = useState(false);
    const [selectedOption, setSelectedOption] = useState(null);
    const [childrenToggle, setChildrenToggle] = useState(false);

    const toggle = () => setIsOpen(!isOpen);

    const onOptionClick = value => () => {
        setSelectedOption(value);
        setIsOpen(false);
    };

    const handleToggle = () => {
        setChildrenToggle(!value)
        setValue(!value)
    }

    return (
        <div className={`${props.flagSize?props.flagSize:"col-md-12"}`}>
            <div className='input-group'>
                <div className='display-flex'>
                    <span className='bullet'/>
                    <span className='label'>{props.name}</span>
                    <div className='input-addon'>
                        {!!props?.hasDropdown && !!props?.dropdown &&
                            <NumericDropdown {...props.dropdown} {...{isOpen, toggle, selectedOption, onOptionClick}} />
                        }
                        <ToggleSwitch name={props.id} id={props.id} value={value} disabled={!!props?.disabled} handleToggle={handleToggle} />
                        {!!props?.hasChildren && !!props?.children?.fields && !!props?.children?.type &&
                            <span className={`childToggle ${childrenToggle ? 'open' : ''}`} onClick={() => setChildrenToggle(!childrenToggle)}></span>
                        }
                    </div>
                </div>
                {!!props?.hasChildren && !!props?.children?.fields && !!props?.children?.type && !!childrenToggle &&
                    <div className={`${props.children.type} row row-no-gutters`}>
                        {props.children.fields.length > 0 &&
                            props.children.fields.map((field, i) => (
                                <InputGroup key={i} {...field} defaultValue={!value ? false : field.defaultValue} disabled={!value?true:field.disabled}/>
                            ))
                        }
                    </div>
                }
            </div>
        </div>
    );
}

export { InputGroup } 