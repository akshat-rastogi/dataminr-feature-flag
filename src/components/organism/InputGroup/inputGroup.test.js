import React from 'react';
import { shallow } from 'enzyme';
import { InputGroup } from './InputGroup';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() });

describe("InputGroup", () => {
    it("should render my component", () => {
        const props = {
            "id": "Alerts.AlertManager",
            "name": "Alert Manager",
            "type": "switch",
            "disabled": false,
            "defaultValue": true,
            "flagSize": ""
        }
        const wrapper = shallow(<InputGroup {...props}/>);
        expect(wrapper.getElements()).toMatchSnapshot();
    });
});