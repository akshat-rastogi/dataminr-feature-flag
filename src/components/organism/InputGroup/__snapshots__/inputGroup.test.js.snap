// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroup should render my component 1`] = `
Array [
  <div
    className="col-md-12"
  >
    <div
      className="input-group"
    >
      <div
        className="display-flex"
      >
        <span
          className="bullet"
        />
        <span
          className="label"
        >
          Alert Manager
        </span>
        <div
          className="input-addon"
        >
          <ToggleSwitch
            disabled={false}
            handleToggle={[Function]}
            id="Alerts.AlertManager"
            name="Alerts.AlertManager"
            value={true}
          />
        </div>
      </div>
    </div>
  </div>,
]
`;
